qiime2/plugin/plugin.py

```diff
@@ -201,9 +205,13 @@ def register_semantic_type_to_format(self, semantic_type, artifact_format):
                 raise ValueError("%r has a predicate, differentiating format"
                                  " on predicate is not supported.")
 
-        self.type_formats.append(TypeFormatRecord(
+        record = TypeFormatRecord(
             type_expression=semantic_type,
-            format=artifact_format, plugin=self))
+            format=artifact_format, plugin=self)
+
+        self.type_formats.append(record)
+        for type_ in record.type_expression:
+            self.importable_types.add(type_)
```

Comment:
Don't forget to update this when you rename `importable_types` to `types`! Also, instead of adding the type to this property, how about you create and add a `SemanticTypeRecord`, instead (I will explain more detail below, in `plugin_manage.py`).